e2e/suite_test.go

```diff
@@ -26,192 +27,195 @@ func Test_All(t *testing.T) {
+		}
+	}
+
+	if err := writeToFile(filepath.Join(caseLogsDir, "vmssId.txt"), *vmssModel.ID); err != nil {
+		t.Fatal("failed to write vmss resource ID to disk", err)
+	}
 
-			vmssModel, err := createVMSSWithPayload(ctx, publicKeyBytes, cloud, suiteConfig.location, vmssName, subnetID, base64EncodedCustomData, cseCmd, tc.vmConfigMutator)
-			isCSEError := isVMExtensionProvisioningError(err)
-			vmssSucceeded := true
+	// Perform posthoc log extraction when the VMSS creation succeeded or failed due to a CSE error
+	if vmssSucceeded || isCSEError {
+		debug := func() {
+			err := pollExtractVMLogs(ctx, t, cloud, kube, suiteConfig, *chosenCluster.Properties.NodeResourceGroup, *vmssModel.Name, caseLogsDir, privateKeyBytes)
 			if err != nil {
-				vmssSucceeded = false
-				if isCSEError {
-					t.Error("VM was unable to be provisioned due to a CSE error, will still atempt to extract provisioning logs...", err)
-				} else {
-					t.Fatal("Encountered an unknown error while creating VM", err)
-				}
+				t.Fatal(err)
 			}
+		}
+		defer debug()
+	}
 
-			if err := writeToFile(filepath.Join(caseLogsDir, "vmssId.txt"), *vmssModel.ID); err != nil {
-				t.Fatal("failed to write vmss resource ID to disk", err)
-			}
+	// Only perform node readiness/pod-related checks when VMSS creation succeeded
+	if vmssSucceeded {
+		t.Log("vmss creation succeded, proceeding with node readiness and pod checks...")
+		if err = validateNodeHealth(ctx, t, kube, *vmssModel.Name); err != nil {
+			t.Fatal(err)
+		}
+		t.Log("node bootstrapping succeeded!")
+	}
+}
 
-			// Perform posthoc log extraction when the VMSS creation succeeded, or failed due to a CSE error
-			if vmssSucceeded || isCSEError {
-				debug := func() {
-					err := wait.PollImmediateWithContext(ctx, 15*time.Second, 5*time.Minute, func(ctx context.Context) (bool, error) {
-						t.Log("attempting to extract VM logs")
-
-						logFiles, err := extractLogsFromVM(ctx, t, cloud, kube, suiteConfig.subscription, vmssName, string(privateKeyBytes))
-						if err != nil {
-							t.Logf("error extracting VM logs: %q", err)
-							return false, nil
-						}
-
-						t.Logf("dumping VM logs to local directory: %s", caseLogsDir)
-						if err = dumpFileMapToDir(caseLogsDir, logFiles); err != nil {
-							t.Logf("error extracting VM logs: %q", err)
-							return false, nil
-						}
-
-						return true, nil
-					})
-					if err != nil {
-						t.Fatal(err)
-					}
-				}
-				defer debug()
-			}
+func bootstrapVMSS(
+	ctx context.Context,
+	t *testing.T,
+	r *mrand.Rand,
+	cloud *azureClient,
+	suiteConfig *suiteConfig,
+	scenario *scenario.Scenario,
+	chosenCluster *armcontainerservice.ManagedCluster,
+	nbc *datamodel.NodeBootstrappingConfiguration,
+	subnetID string,
+	publicKeyBytes []byte) (*armcompute.VirtualMachineScaleSet, func(), error) {
+	nodeBootstrapping := mustGetNodeBootstrapping(ctx, t, nbc)
+
+	vmssName := fmt.Sprintf("abtest%s", randomLowercaseString(r, 4))
+	t.Logf("vmss name: %q", vmssName)
```

Comment:
nit: I suggest avoiding using the test helpers explicitly in the deeper nested logic, use fmt.Errorf and function wrappers to return errors up the stack to the main test func, and we can start plumbing a propper logger down the stack (in func signatures)

this will help @utheman with integrationt test coverage as well

Reply:
removed from here - will be nice to have a separate logger implementation to use so we don't have to plumb 't' around everywhere

Reply:
you can fmt.Println already or log.Println, t.Log is nothing fancy

t.Error/Fatal are just println/and t.fail/failnow()